src/ansys/fluent/visualization/graphics/graphics_windows_manager.py

```diff
@@ -620,7 +632,11 @@ def open_window(
         with self._condition:
             if not window_id:
                 window_id = self._get_unique_window_id()
-            if in_notebook() or get_config()["blocking"] or pyviz.SINGLE_WINDOW:
+            if (
+                in_notebook()
+                or not pyviz.config.interactive
+                or pyviz.config.single_window
+            ):
```

Comment:
- This condition is repeated four times. 
- The internal logic is opaque, and its connection to the resulting actions is unclear.
- It is unclear if the methods with names ending `_notebook` have something to do with jupyter notebook or something else. 
- This repetition, based on fixed configuration flags, in a single class is a clear case for polymorphism. We shall extract the branching logic into separate classes or strategies to clarify intent and eliminate duplication.

Reply:
Will continue this as future tasks.